pycroft/model/task_serialization.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from datetime import date
+from marshmallow import Schema, fields, post_load
+
+
+class UserMoveOutSchema(Schema):
+    comment = fields.Str()
+    end_membership = fields.Bool()
+
+    @post_load
+    def build(self, data, **kwargs):
+        return UserMoveOutParams(**data)
+
+
+@dataclass
+class UserMoveOutParams:
+    comment: str
+    end_membership: bool
+
```

Comment:
The `*Schema` and `*Params` classes cannot be combined with marshmallow? Because this seems to be a little bit unnecessary. [Pydantic](https://pydantic-docs.helpmanual.io/) for example uses the python types for validating and you only need one class per Schema.

Reply:
Doesn't seem that way, but there is [marshmallow-dataclass](https://pypi.org/project/marshmallow-dataclass/).
I'd say we can think about this in the future, but in this refactoring PR I don't want to introduce a new library.

Reply:
It's also not that bad because the outer code does not need to use both the class and the schema: Once `Task.params` gets assigned a value, the required schema is inferred from the `TaskType` and dumping is initiated.
The „end-user“ only needs to import e.g. `UserMoveParams`, and not `UserMoveSchema`.
Resolving.